packages/eventual-send/src/index.js

```diff
@@ -448,7 +449,7 @@ export function makeHandledPromise(Promise) {
         HandledPromise.resolve()
           .then(() =>
             // and resolve to the answer from the specific unfulfilled handler,
-            resolve(unfulfilledHandler[operation](p, ...args)),
+            resolve(unfulfilledHandler[operation](p, ...args, returnedP)),
```

Comment:
That's a splat, right? So the handler's operation function will get anywhere from 2 to N+2 arguments, depending upon `args.length`?

I'm not sure where this particular `...args` comes from.. do we need to worry that the caller and the receiver might not agree on the number of arguments, and then `returnedP` winds up in the wrong place? Most of the time I see varargs, they strictly come at the end of the function to avoid such confusion. But if our own code is on both ends of this call, maybe we don't need to worry about it.

Reply:
I second what Brian said.  That looks like an accident waiting to happen.

Reply:
Yes, our own code is at both ends.

Reply:
I understand why it looks confusing but it is the best way to write this. However, the variable name `args` perhaps should be renamed `metaArgs` or something, to indicate that's it at a metalevel compared to the normal use of `args` for an array of arguments. For example, then the `operation` is `applyMethod`, the contents of, let's say, `metaArgs` is `[prop, args]`, where `prop` is the name of the remote method to be called and `args` are the arguments to be spread out and passed to it.

In explaining this, it occurs to me that `prop` vs `operation` has the same meta-level shift as `args` vs `metaArgs`.

Reply:
I added a comment to this effect.